libraries/botbuilder-ai/src/qnamaker-utils/httpRequestUtils.ts

```diff
@@ -45,14 +45,20 @@ export class HttpRequestUtils {
 
         const headers: any = this.getHeaders(endpoint);
 
-        const qnaResult: any = await fetch(requestUrl, {
-            method: 'POST',
-            headers: headers,
-            timeout: timeout,
-            body: payloadBody
-        });
-        
-        return await qnaResult.json();
+        try {
+            const qnaResult = await fetch(requestUrl, {
+                method: 'POST',
+                headers: headers,
+                timeout: timeout,
+                body: payloadBody
+            });
+            const jsonResult = qnaResult.status == 204 ? this.getSuccessfulTrainApiResult() : await qnaResult.json();
```

Comment:
@Zerryth @vipeketi, can we validate the requestUrl ahead of time to know that we're specifically calling the Train API? Do other APIs return 204 status codes?

Reply:
Hmm, as far as I know, Train API was the only one that returns 204, but I say that with a grain of salt, as I think there may be more newer qna APIs that I haven't gotten a chance to play with quite yet.

However even if it is just the Train API for now that returns no-content, I think I see what you're getting at, and I'll rework this to verify that they're calling the Train API 1st, since maybe in the future some other new qna API might return no-content, and it would be wrong to return a qna result that said that training was successful, if it wasn't that very API we called

Reply:
@stevengum delete, update, replace  APIs return 204 status codes.. Guess, GenerateAnswer and Train API are more relevant to the context .. Handling TrainAPI for 204 should suffice for current situation.

Reply:
Updated code (again) to handle more than just specifically the Train API when a 204 response occurs.
This is better for future proofing as well, since who knows if in the future if QnA will include more APIs that respond with 204

...ok I'm going to bed 😅 
lmk if we want to go a diff direction